optuna/samplers/_partial_fixed.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any
+from typing import Dict
+import warnings
+
+import numpy
+
+from optuna._experimental import experimental
+from optuna.distributions import BaseDistribution
+from optuna.samplers import BaseSampler
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+
+
+@experimental("2.3.0")
+class PartialFixedSampler(BaseSampler):
+    """Sampler that can sample parameters with fixed parameters.
+
+        .. versionadded:: 2.3.0
+
+    Example:
+
+         After several steps of optimization, you can fix the value of ``y`` and re-optimize it.
+
+        .. testcode::
+
+            import optuna
+
+
+            def objective(trial):
+                x = trial.suggest_uniform("x", -1, 1)
+                y = trial.suggest_int("y", -1, 1)
+                return x ** 2 + y
+
+
+            study = optuna.create_study()
+            study.optimize(objective, n_trials=10)
+
+            best_params = study.best_params
+            fixed_params = {"y": best_params["y"]}
+            partial_sampler = PartialFixedSampler(fixed_params, study.sampler)
+
+            study.sampler = partial_sampler
+            study.optimize(objective, n_trials=10)
+
+    """
+
+    def __init__(self, fixed_params: Dict[str, Any], base_sampler: BaseSampler) -> None:
+        self._fixed_params = fixed_params
+        self._base_sampler = base_sampler
+        self._rng = numpy.random.RandomState()
```

Comment:
I guess `self._rng` is not used in the sampling procedure (i.e., `infer_relative_search_space`, `sample_relative` and `sample_independent`). Could you tell me why we need it?

Reply:
Thank you @toshihikoyanase.
As you pointed out self._rng was unnecessary, so i fixed it.